R/parse.R

```diff
@@ -116,23 +116,25 @@ add_id_and_short <- function(pd) {
 }
 
 
-#' Ensure a correct `text` of all strings
+#' Ensure a correct `text` of all strings and numeric constants
 #'
-#' Make sure `text` of the tokens `STR_CONST` is correct and adapt if necessary.
-#' We first parse `text` again and include also non-terminal text. Then, we
-#' replace offending `text` in the terminal expressions with the text of their
-#' parents if their line / col position matches and return an error otherwise.
+#' Make sure `text` of the tokens `STR_CONST` and `NUM_CONST` is correct and
+#' adapt if necessary. We replace offending `text` in the terminal expressions
+#' with the text of their parents if their line / col position matches and
+#' return an error otherwise.
 #' @param pd A parse table.
-#' @param text The text from which `pd` was created. Needed potentially
-#'   for another round of parsing.
 #' @importFrom rlang abort
+#' @importFrom magrittr or
 #' @keywords internal
-ensure_correct_str_txt <- function(pd, text) {
+ensure_correct_txt <- function(pd, text) {
   ensure_valid_pd(pd)
-  is_problematic_string <- identify_insufficiently_parsed_strings(pd, text)
-  problematic_strings <- pd[is_problematic_string, ]
+  is_problematic_string <- or(
+    is_insufficiently_parsed_strings(pd),
+    is_insufficiently_parsed_num_const(pd)
+  )
+  problematictext <- pd[is_problematic_string, ]
   is_parent_of_problematic_string <-
-    pd$id %in% problematic_strings$parent
+    pd$id %in% problematictext$parent
```

Comment:
Underscores have gone missing here, is this intended?

Reply:
Nope, reverted.